server/plugin/webhook.go

```diff
@@ -1377,3 +1385,44 @@ func (p *Plugin) postStarEvent(event *github.StarEvent) {
 		}
 	}
 }
+
+func (p *Plugin) postWorkflowJobEvent(event *github.WorkflowJobEvent) {
+	if event.GetAction() != actionCompleted {
+		return
+	}
+
+	// Create a post only when the workflow job is completed and fails
+	if event.GetWorkflowJob().GetConclusion() != workflowJobFail {
+		return
+	}
+
+	repo := event.GetRepo()
+	subs := p.GetSubscribedChannelsForRepository(repo)
+
+	if len(subs) == 0 {
+		return
+	}
+
+	newWorkflowJobMessage, err := renderTemplate("newWorkflowJob", event)
+	if err != nil {
+		p.client.Log.Warn("Failed to render template", "Error", err.Error())
+		return
+	}
+
+	for _, sub := range subs {
+		if !sub.Workflows() {
+			continue
+		}
```

Comment:
Makes sense to me :+1: 

Limiting the notifications to a specific branch seems like a common use case. Can we include that in there as well?

Reply:
> `Just the ci workflow`

@mickmister Are you suggesting here to add a subscription on the basis of the workflow name, which is `ci` in the above case?

Reply:
> Limiting the notifications to a specific branch seems like a common use case. Can we include that in there as well?

@mickmister @hanzei For the above case, what should be the slash command format? Should we keep something like: `/github subscriptions add owner/repo pulls,workflows:branch-b1,b2;name-n1,n2;job-j1,j2`? Please let me know if you have any better suggestions for the above case.

Reply:
The `subscriptions add` feature is quite overloaded at this point. Adding more complexity on top of it may not be the best thing to do. Created a ticket for breaking this out of the command https://github.com/mattermost/mattermost-plugin-github/issues/780

Mainly thinking of the complexity of the `job` parameter there. I'm thinking we should avoid the `job` parameter, since those jobs may not belong to all workflows. Then it's up to the actions workflow developer in that project to structure the workflows in a way to be something to subscribe to.